packages/formstr-app/src/containers/CreateFormNew/components/FormDetails/index.tsx

```diff
@@ -64,9 +83,62 @@ export const FormDetails: React.FC<FormDetailsProps> = ({
     setSavedLocally(true);
   };
 
+  const saveToMyForms = async (
+    formAuthorPub: string,
+    formAuthorSecret: string,
+    formId: string,
+    relay: string,
+    viewKey?: string
+  ) => {
+    if (!userPub) return;
+    setSavingOnNostr(true);
+    let existingListFilter = {
+      kinds: [14083],
+      authors: [userPub],
+    };
+    let pool = new SimplePool();
+    let existingList = await pool.querySync(
+      getDefaultRelays(),
+      existingListFilter
+    );
+    let forms: Tag[] = [];
+    if (existingList[0]) {
+      let formsString = await window.nostr.nip44.decrypt(
+        userPub,
+        existingList[0].content
+      );
+      forms = JSON.parse(formsString);
+    }
+    let key = `${formAuthorPub}:${formId}`;
+    if (forms.map((f) => f[1]).includes(key)) {
+      setSavedOnNostr(true);
```

Comment:
can probably use a single state called `state`. it can be `null`, `saving`, `saved` etc